### DISCUSSION TITLE
How can I focus a TextInput programmatically?

### DISCUSSION THREAD

**Discussion by geoidesic** [carbon-design-system/carbon-components-svelte#1799] [Q&A] [answered]
https://carbondesignsystem.com/components/text-input/usage/

According to the docs, the only way is using tab. 

Normally with an input I would do:
```
<input bind:this={inputEl} />
```

Then inputEL has a focus method that I can call to focus the element. But with TextInput, it doesn't work:
```
<TextInput bind:this={inputEl} />
```
In this case inputEL is a svelte component, not an HTMLelement, so it doesn't have the focus method.

So how can I focus the text input from the parent?

**Comment by metonym** [comment, marked as answer]
```svelte
<script>
  import { TextInput } from "carbon-components-svelte";

  let ref;
</script>

<TextInput bind:ref />

<button on:click={() => ref?.focus()}>
  Focus
</button>
```
